projects/cli-files/go-ls/cmd/root.go

```diff
@@ -0,0 +1,86 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+)
+
+func Execute() error {
+	pathArr := os.Args[1:]
+
+	var path string
+	var comma bool
+
+	if len(pathArr) == 0 {
+		path = "."
+		directoryOutput(path, comma)
+		return nil
+	}
+
+	if pathArr[0] == "-m" {
+		comma = true
+	} else {
+		comma = false
+	}
+
+	if len(pathArr) == 1 && comma {
+		path = "."
+		directoryOutput(path, comma)
+		return nil
+	}
+
+	var i int
+	if comma {
+		i = 1
+	} else {
+		i = 0
+	}
+
+	for i < len(pathArr) {
+
+		isPathValid, err := os.Stat(pathArr[i])
+		if err != nil {
+			return fmt.Errorf("%s no such file or directory", pathArr[i])
+		}
+
+		isDirectory := isPathValid.IsDir()
+		if !isDirectory {
+			if comma && i < len(pathArr)-1 {
+				os.Stdout.Write([]byte(pathArr[i] + ", "))
+			} else {
+				os.Stdout.Write([]byte(pathArr[i] + "  "))
+			}
+		} else {
+			dirPath1 := len(pathArr) > 1 && !comma
+			dirPath2 := len(pathArr) > 2 && comma
```

Comment:
I'm not sure from their names what `dirPath1` and `dirPath2` are meant to mean - can you think of names that make this more clear?

(Or maybe this goes away if you switch how you populated `pathArr` above, so that rather than needing to worry about whether some of the things in `pathArr` are not paths, you end up with `pathArr` always containing just the list of paths, so not needing to worry about `comma` when doing this check :))